quinn/src/endpoint.rs

```diff
@@ -358,3 +351,54 @@ impl Drop for Incoming {
         }
     }
 }
+
+pub(crate) struct EndpointRef(Arc<Mutex<EndpointInner>>);
+
+impl EndpointRef {
+    pub(crate) fn new(log: Logger, socket: UdpSocket, inner: quinn::Endpoint, ipv6: bool) -> Self {
+        let (sender, events) = mpsc::unbounded();
+        let x = Self(Arc::new(Mutex::new(EndpointInner {
```

Comment:
Why do we need the intermediate variable here?

Reply:
Debugging leftovers, good catch.